app/scripts/controllers/metamask-notifications/metamask-notifications.ts

```diff
@@ -395,28 +477,118 @@ export class MetamaskNotificationsController extends BaseController<
+   * to show or hide loading indicators in the UI when notifications are being fetched.
+   *
+   * @param isFetchingMetamaskNotifications - A boolean value representing the fetching state.
    */
+  #setIsFetchingMetamaskNotifications(
+    isFetchingMetamaskNotifications: boolean,
+  ) {
+    this.update((s) => {
+      s.isFetchingMetamaskNotifications = isFetchingMetamaskNotifications;
+    });
+  }
+
+  /**
+   * Updates the state to indicate that the checking of accounts presence is in progress.
+   *
+   * This method modifies the `isCheckingAccountsPresence` state, which can be used to manage UI elements
+   * that depend on the status of account presence checks, such as displaying loading indicators or disabling
+   * buttons while the check is ongoing.
+   *
+   * @param isCheckingAccountsPresence - A boolean value indicating whether the account presence check is currently active.
+   */
+  #setIsCheckingAccountsPresence(isCheckingAccountsPresence: boolean) {
+    this.update((s) => {
+      s.isCheckingAccountsPresence = isCheckingAccountsPresence;
+    });
+  }
+
+  /**
+   * Updates the state to indicate that account updates are in progress.
+   * Removes duplicate accounts before updating the state.
+   *
+   * @param accounts - The accounts being updated.
+   */
+  #updateUpdatingAccountsState(accounts: string[]) {
+    this.update((s) => {
+      const uniqueAccounts = new Set([
+        ...s.isUpdatingMetamaskNotificationsAccount,
+        ...accounts,
+      ]);
+      s.isUpdatingMetamaskNotificationsAccount = Array.from(uniqueAccounts);
+    });
+  }
+
+  /**
+   * Clears the state indicating that account updates are complete.
+   *
+   * @param accounts - The accounts that have finished updating.
+   */
+  #clearUpdatingAccountsState(accounts: string[]) {
+    this.update((s) => {
+      s.isUpdatingMetamaskNotificationsAccount =
+        s.isUpdatingMetamaskNotificationsAccount.filter(
+          (existingAccount) => !accounts.includes(existingAccount),
+        );
+    });
+  }
+
   public async checkAccountsPresence(
     accounts: string[],
   ): Promise<Record<string, boolean>> {
-    // Retrieve user storage
-    const userStorage = await this.#getUserStorage();
-    this.#assertUserStorage(userStorage);
+    try {
+      this.#setIsCheckingAccountsPresence(true);
 
-    // Use MetamaskNotificationsUtils to check the presence of accounts
-    return MetamaskNotificationsUtils.checkAccountsPresence(
-      userStorage,
-      accounts,
-    );
+      // Retrieve user storage
+      const userStorage = await this.#getUserStorage();
+      this.#assertUserStorage(userStorage);
+
+      const presence = MetamaskNotificationsUtils.checkAccountsPresence(
+        userStorage,
+        accounts,
+      );
+
+      this.#setIsCheckingAccountsPresence(false);
+
+      return presence;
+    } catch (error) {
+      this.#setIsCheckingAccountsPresence(false);
+      throw error;
+    }
```

Comment:
Can `this.#setIsCheckingAccountsPresence(false);` be put into a `finally` instead of both spots?

Reply:
100%! Fixed